From: Zhang Zhen <zhenzhang.zhang@huawei.com>
Date: Thu, 10 Apr 2014 13:29:17 +0800
Subject: [PATCH] ceph: remove useless ACL check
Git-commit: 979d4c1895474aafa473f441846dd3b6f7276202
Patch-mainline: v3.16
References: Fate#318586


posix_acl_xattr_set() already does the check, and it's the only
way to feed in an ACL from userspace.
So the check here is useless, remove it.

Signed-off-by: zhang zhen <zhenzhang.zhang@huawei.com>
Reviewed-by: Yan, Zheng <zheng.z.yan@intel.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/acl.c |    6 ------
 1 file changed, 6 deletions(-)

--- a/fs/ceph/acl.c
+++ b/fs/ceph/acl.c
@@ -111,12 +111,6 @@ static int ceph_set_acl(struct dentry *d
 	struct iattr newattrs;
 	umode_t new_mode = inode->i_mode, old_mode = inode->i_mode;
 
-	if (acl) {
-		ret = posix_acl_valid(acl);
-		if (ret < 0)
-			goto out;
-	}
-
 	switch (type) {
 	case ACL_TYPE_ACCESS:
 		name = POSIX_ACL_XATTR_ACCESS;
